app/controllers/application_controller.rb

```diff
@@ -65,7 +65,6 @@ def self.action_auth_level(action, level)
 
     if level == :administrator
       skip_before_action :authorize_user_for_course, only: [action], raise: false
-      skip_before_action authenticate_for_action: [action]
```

Comment:
Currently, this line of code is silently ignored because `authenticate_for_action` isn't defined at this point.

Even so, this line of code should be removed. Otherwise, we are skipping authentication and allowing non-admins to access admin routes.

For example, if we change this line to `skip_before_action :authenticate_for_action, only: [action], raise: false`, we observe that non-admins can access `/admin/github_integration` and `/admin/email_instructors` routes.